### PR TITLE
HOTFIX: StateDirectoryTest should use Set instead of List

### PR DESCRIPTION
**Pull request by abbccdda**
This PR fixes the undefined order of list directory result, to make the test result consistent, otherwise we would hit out of order from time to time:

```
java.lang.AssertionError: expected:<[/tmp/kafka-k0Nn0/applicationId/0_0, /tmp/kafka-k0Nn0/applicationId/0_1]> but was:<[/tmp/kafka-k0Nn0/applicationId/0_1, /tmp/kafka-k0Nn0/applicationId/0_0]>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.apache.kafka.streams.processor.internals.StateDirectoryTest.shouldOnlyListNonEmptyTaskDirectories(StateDirectoryTest.java:318)
```

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by mjsax — APPROVED** [review]
LGTM. Thanks for the quick fix!

**Comment by mjsax** [comment]
Retest this please.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java [hunk 1]
```diff
@@ -315,13 +322,17 @@ public void shouldOnlyListNonEmptyTaskDirectories() {
...
+        assertEquals(mkSet(taskDir1), Arrays.stream(
+            directory.listNonEmptyTaskDirectories()).collect(Collectors.toSet()));
 
         directory.cleanRemovedTasks(0L);
 
-        assertEquals(Arrays.asList(taskDir1, taskDir2), Arrays.asList(directory.listAllTaskDirectories()));
-        assertEquals(Collections.emptyList(), Arrays.asList(directory.listNonEmptyTaskDirectories()));
+        assertEquals(mkSet(taskDir1, taskDir2), Arrays.stream(
+            directory.listAllTaskDirectories()).collect(Collectors.toSet()));
+        assertEquals(Collections.emptySet(), Arrays.stream(
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java:334]: Good call.

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Comment by abbccdda** [comment]
Btw, I guess the build won't pass as current trunk has another breakage caused by request quota, will wait and see when that one gets resolved @guozhangwang @mjsax

**Comment by mjsax** [comment]
Retest this please

**Comment by mjsax** [comment]
Seems the `push -f` killed the build -- not sure why I could not restart it -- also, I rebased to `trunk` recently and it worked -- seems all fixes got merged already?

**Comment by abbccdda** [comment]
@mjsax The fix is merged

**Comment by mjsax** [comment]
Cool. Build is running now. :)

**Pull request merged**